src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -270,7 +270,7 @@ private async Task RenegotiateAsync<TIOAdapter>(TIOAdapter adapter)
 
             try
             {
-                if (_decryptedBytesCount is not 0)
+                if (_decryptedBytesCount != 0 || _internalBufferCount != 0)
```

Comment:
Nit: maybe:
```suggestion
                if ((_decryptedBytesCount | _internalBufferCount) != 0)
```
?

Reply:
Why? Better perf?

Reply:
I would have hoped that the compiler/JIT would do this for me

Reply:
> Why? Better perf?

Slightly, yes.  One branch instead of two.

> I would have hoped that the compiler/JIT would do this for me

@JulieLeeMSFT added some such optimizations in https://github.com/dotnet/runtime/pull/49548, but I don't believe this case is covered.